consensus/core/src/dag_state.rs

```diff
@@ -58,11 +61,19 @@ pub(crate) struct DagState {
 
     // Persistent storage for blocks, commits and other consensus data.
     store: Arc<dyn Store>,
+
+    // The number of cached rounds
+    cached_rounds: Round,
 }
 
 impl DagState {
     /// Initializes DagState from storage.
-    pub(crate) fn new(context: Arc<Context>, store: Arc<dyn Store>) -> Self {
+    pub(crate) fn new(
+        context: Arc<Context>,
+        store: Arc<dyn Store>,
+        cached_rounds: Option<Round>,
```

Comment:
We could have it in Parameters yes, but I feel is more like an internal setting rather than something we would like to allow validators to set them selfs (especially given assumptions that we make in the code).